core/src/scene/importer.cpp

```diff
@@ -0,0 +1,251 @@
+#include "importer.h"
+#include "platform.h"
+#include "scene/sceneLoader.h"
+#include "util/topologicalSort.h"
+#include "yaml-cpp/yaml.h"
+
+#include <regex>
+
+using YAML::Node;
+using YAML::NodeType;
+
+namespace Tangram {
+
+Node Importer::applySceneImports(const std::string& scenePath) {
+
+    if (loadScene(scenePath)) {
+        return importScenes(scenePath);
+    }
+
+    return Node();
+}
+
+std::string Importer::getFilename(const std::string& filePath) {
+    std::string sceneName = filePath;
+    std::regex r("[^//]+$");
+    std::smatch match;
+    if (std::regex_search(sceneName, match, r)) { sceneName = match[0]; }
+    return sceneName;
+}
+
+std::string Importer::normalizePath(const std::string &path,
+                                         const std::string &parentPath) {
+
+    //TODO: Handle network urls
+
+    if (path[0] == '/') {
+        //absolute path, return as it is
+        return path;
+    }
+
+    std::regex r("[^//]+$");
+    return std::regex_replace(parentPath, r, path);
+}
+
+void Importer::normalizeSceneImports(Node& root, const std::string& parentPath) {
+    if (Node import = root["import"]) {
+        if (import.IsScalar()) {
+            import = normalizePath(import.Scalar(), parentPath);
+        } else if (import.IsSequence()) {
+            for (Node imp : import) {
+                if (imp.IsScalar()) { imp = normalizePath(imp.Scalar(), parentPath); }
+            }
+        }
+    }
+}
+
+void Importer::setNormalizedTexture(Node& texture, const std::vector<std::string>& names,
+        const std::string& parentPath) {
+
+    for (auto& name : names) {
+        if (m_textureNames.find(name) == m_textureNames.end()) {
+            m_textureNames.insert(name);
+            if (names.size() > 1) {
+                texture.push_back(normalizePath(name, parentPath));
```

Comment:
It should not `push_back` on the container here, we should access `texture[current_index]` and update it instead.